service/namespace_service.go

```diff
@@ -52,6 +53,28 @@ func (s namespaceService) Get(ctx context.Context, projectName string, namespace
 	return nsSpec, nil
 }
 
+// GetOptional is used for optionally getting namespace if name is present, otherwise get only project
+func (s namespaceService) GetOptional(ctx context.Context, projectName string, namespaceName string) (models.NamespaceSpec, error) {
```

Comment:
Return both project and namespace, and name should reflect the namespace can be optional